dd-sdk-android/src/main/kotlin/com/datadog/android/log/internal/logger/LogcatLogHandler.kt

```diff
@@ -70,22 +70,30 @@ internal class LogcatLogHandler(
     internal fun getCallerStackElement(): StackTraceElement? {
         return if (Datadog.isDebug && useClassnameAsTag) {
             val stackTrace = Throwable().stackTrace
-            stackTrace.firstOrNull {
-                it.className !in ignoredClassNames
-            }
+            return findValidCallStackElement(stackTrace)
         } else {
             null
         }
     }
 
+    internal fun findValidCallStackElement(
+        stackTrace: Array<StackTraceElement>
+    ): StackTraceElement? {
+        return stackTrace.firstOrNull { element ->
+            element.className !in IGNORED_CLASS_NAMES &&
+                IGNORED_PACKAGE_PREFIXES.none { element.className.startsWith(it) }
+        }
+    }
+
     // endregion
 
     companion object {
```

Comment:
is there any reason `companion` is public? there is no usage of this object in the code anyway.

on another side making it `internal` doesn't help, because `internal` symbols are visible from java side and the only solution to hide it is to either use `JvmName` with java-prohibited style or use `JvmSynthetic`, which is not yet used in the project.

Reply:
nope, there's no special reason for that.